ludwig/modules/metric_modules.py

```diff
@@ -109,7 +110,11 @@ def sync_context(
         distributed_available: Optional[Callable] = jit_distributed_available,
     ) -> Generator:
         """Override the behavior of this in the base class to support custom distributed strategies."""
-        dist_strategy = get_current_dist_strategy()
+        try:
+            dist_strategy = get_current_dist_strategy()
+        except RuntimeError:
+            # We don't initialize distributed strategy for GBM models. Default to LocalStrategy.
+            dist_strategy = LocalStrategy()
```

Comment:
I don't like this solution. The idea behind making the distributed strategy explicit is that we should always be setting it. The problem with this approach is that it allows us to accidentally forget to set it, then we get unexpected behavior. I would prefer we find out why we're not setting the distributed strategy for GBMs when we should be.